Sources/Purchasing/ReceiptFetcher.swift

```diff
@@ -151,10 +151,11 @@ private extension ReceiptFetcher {
         sleepDuration: DispatchTimeInterval
     ) async -> Data {
         var retries = 0
+        var data: Data = .init()
 
         repeat {
             retries += 1
-            let data = await self.refreshReceipt()
+            data = await self.refreshReceipt()
```

Comment:
When the receipt can't be parsed on the last retry, we would return that receipt. Is that correct?

Reply:
We could add a couple of tests for these returning an unparseable receipt and an empty receipt if that's the case.

Reply:
> When the receipt can't be parsed on the last retry, we would return that receipt. Is that correct?

Correct, as explained in the description:
> To make this retry mechanism less risky when eventually enabled, this change makes the "worst case scenario" [...] equivalent to `ReceiptFetcher.onlyIfEmpty`.

Reply:
> The current tests already cover that though?

Well I think we have tests for receipts without a purchase but not for unparseable receipts? (I might be missing something since I'm not that familiar with the parsing logic)

Reply:
Is `testStopsRetryingEvenIfDataIsInvalid` what you're describing?

Reply:
```
    func testStopsRetryingEvenIfDataIsInvalid() async {
        self.mock(receipt: Self.receiptWithoutPurchases)
        let invalidData = Self.receiptWithoutPurchases.asData

        let data = await self.fetch(productIdentifier: Self.productID, retries: 2)
        expect(data) == Data()
        expect(data) == invalidData

        expect(self.mockReceiptParser.invokedParseParametersList) == [
            invalidData,
            invalidData,
            invalidData
        ]
        expect(self.mockRequestFetcher.refreshReceiptCalledCount) == 3
    }
```

That seems to be using a `receiptWithoutPurchases`. I was thinking more along the lines of a receipt that is missing a critical field or something like that. So basically a test where we make `self.receiptParser.parse` throw an exception.